### ISSUE TITLE
Elementum 0.1.77 seasons will not download properly anymore

### ISSUE THREAD

**Issue by ThurahT** [elgatito/plugin.video.elementum#759]
Since the last update to 0.1.77 torrents with seasons will not start with downloading the first episode like before. In fact it downloads everything except the first episode and then times out since it reaches the timelimit for the first episode, which never got downloaded. Single episodes torrents are no problem.

If I then exit kodi and restart it elmentum/burst will then resume the download and fetch the missing episode and I can then watch from the first episode.

Edit: scratch that. Season downloading is broken. It keeps downloading every episode except the one I am trying to watch.

**Comment by antonsoroko** [comment]
Log file?
(Link how to make it was in issue template)

**Comment by antonsoroko** [comment]
Also what is you settings regarding download season packs? (E.g. single file/season files/all files).

**Comment by ThurahT** [comment]
I usually download whole seasons. So I guess "files/all files ". I actually found a crash log from yesterday that contains all what I described in my previous post. It was quite infuriating since the episodes I wanted to see was the ones that refused being downloaded.

The torrent section said 100% but if I manually play/resume it jumped down to 45% which seemed more true. But did it ever download the whole thing? Nope.

Just tried with the TV/computer in the living room which is still waiting for the last upgrades and no issues there. Works as intended.

[kodi_crashlog-20210521_034242.log](https://github.com/elgatito/plugin.video.elementum/files/6525365/kodi_crashlog-20210521_034242.log)

I guess from line 426 and onwards.
Hope it helps! I only use elementum and it has beeen the best. Hands down.

**Comment by antonsoroko** [comment]
@ThurahT 
i meant value of setting called "Which files to download" in elementum.

if i am not mistaken, in log i see that you use "Season files" - please confirm this by opening elementum settings in kodi.

"Season files" should download selected file (e.g. s02e01) *and* all other files belonging to season (e.g. s02eXX).
and you have "Use files" as storage, that is suitable for "Season files".

also would be easier if you would stick to issue template and provided steps to reproduce your issue.
specifically `also how exactly you start download of season pack?`
in log i can see that you run search for episode s02e04 from trakt show progress and then choose season pack. 

so i test the torrent you used with the settings like yours - it works fine.
of course it takes much more time to start playback since you said to elementum to download all files of the season.
but with my connection speed elementum is still able to finish buffering before the timeout.

so, to solve your particular issue, you can:

1. set "Which files to download" to "Only playback file"
2. increase "Initial buffering timeout (seconds)" (60s by default)
3. increase your internet speed

**Comment by ThurahT** [comment]
> @ThurahT
> i meant value of setting called "Which files to download" in elementum.
>
I had it to playback then I changed that setting around hoping it would fix stuff. 
> if i am not mistaken, in log i see that you use "Season files" - please confirm this by opening elementum settings in kodi.
> 
Possible. It doesn't matter which I choose. See below.
> "Season files" should download selected file (e.g. s02e01) _and_ all other files belonging to season (e.g. s02eXX).
> and you have "Use files" as storage, that is suitable for "Season files".
> 
> also would be easier if you would stick to issue template and provided steps to reproduce your issue.
Steps:
1: Choose a show.
2: Select a full season
3: Realize 0.0.77 misbehaves.
> specifically `also how exactly you start download of season pack?`
Well, I choose a torrent that is a full season of course. As I have done numerous times before. Then it starts with the first episode and I can usually watch it within 10 seconds of buffering. By the time I've done watching the rest of the season is waiting for me.
> in log i can see that you run search for episode s02e04 from trakt show progress and then choose season pack.
> 
> so i test the torrent you used with the settings like yours - it works fine.
> of course it takes much more time to start playback since you said to elementum to download all files of the season.
> but with my connection speed elementum is still able to finish buffering before the timeout.
> 
> so, to solve your particular issue, you can either:
> 
>     1. set "Which files to download" to "Only playback file"
> 
It was set to playback file. Not all files. But it doesn't matter if I choose all files or season or playback. It doesn't start with first/next episode as it used to do before 0.0.77. It just sits there and downloads half of every other file and then tells me 100%. Which is not even true.
>     2. increase "Initial buffering timeout (seconds)" (60s by default)
> 
I have it set to 60 seconds. I have never even come close to that because before the download started with first episode and I could start watching after ten seconds.
>     3. increase your internet speed
I think 100/100 is fine.

**Comment by antonsoroko** [comment]
then i dunno.
i do not see anything that could lead to this issue in 0.1.77 https://github.com/elgatito/elementum/commits/master
also it does work for me with the same torrent that you used.

maybe author will be able to help.

**Comment by elgatito** [comment]
@ThurahT There is nothing to discuss without a proper log file, with enabled debug log level in Elementum (last settings tab).

**Comment by ThurahT** [comment]
@elgatito I'll do better next time I file a bug report. I didn't now there were an additional level to the logging. Or, I guess I did, but I forgot a long the way.

Anyway, I pretty sure I managed to fix it. I guess I should have collected the proper logs before hand and then after, but I was just testing things out, and what I did was just clearing all caches and old torrents and restart kodi.

Now seasons work as usual again. I have files set to "playback" and that works as intended. So does "all files" and "seasons" as well. So, I guess I just need to clean out caches and old torrents from time to time. Or between releaeses at least.

I consider Elementum to be the best kodi addon I ever used since I started using it back in I don't know. 2016-17? And I am very happy with its continued existence with patching and bug fixes.

I'm sorry for the noise with this issue I made. I consider it solved and closed.
